backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopHighLevelCompiler.scala

```diff
@@ -137,6 +137,10 @@ final class BloopHighLevelCompiler(
               val msg = "Encountered a StackOverflowError coming from the compiler. You might need to restart your Bloop build server"
               logger.error(s"${msg}:\n${t.getStackTrace().mkString("\n")}")
               throw new CompileFailed(new Array(0), msg, new Array(0), t)
+            case t: NoClassDefFoundError =>
+              val msg = "Encountered a NoClassDefFoundError coming from the compiler. You might need to clean compile your workspace"
```

Comment:
I wonder, can bloop recover from such error by doing clean compile itself?

Reply:
So on one hand yes, but on the other I am not able to cover that with a test or reproduce it, so I can't guarantee it will work the way we want :sweat: 

This also should most likely be fixed in the incremental compilation, so ideally this never happens in the future.

Reply:
So ideally this should be up to the user. We will show error message in such a case in Metals, so we might add buttons for clean compile and restart server maybe?

Reply:
that sounds reasonable